docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_publicly_accessible_remediation.mdx

```diff
@@ -1,6 +1,213 @@
+4. **Verify the Configuration**: Optionally, verify the cluster's configuration to ensure it is not publicly accessible.
+
+Replace `'YOUR_AWS_ACCESS_KEY'`, `'YOUR_AWS_SECRET_KEY'`, `'YOUR_AWS_REGION'`, and `'your-cluster-identifier'` with your actual AWS credentials, region, and Redshift cluster identifier.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the Amazon Redshift console at https://console.aws.amazon.com/redshift/.
+
+2. In the navigation pane, choose "Clusters". This will display a list of all your Redshift clusters.
+
+3. Click on the name of the cluster that you want to check. This will open the cluster details page.
+
+4. In the cluster details page, look for the "Publicly accessible" field under the "Network and security" section. If the value of this field is "Yes", then the Redshift cluster is publicly accessible.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local system. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all Redshift clusters: Use the following AWS CLI command to list all the Redshift clusters in your account:
+
+   ```
+   aws redshift describe-clusters
+   ```
+
+   This command will return a JSON output with details of all the Redshift clusters.
+
+3. Check the PubliclyAccessible attribute: In the JSON output, look for the "PubliclyAccessible" attribute for each cluster. If the value of this attribute is "true", then the Redshift cluster is publicly accessible. 
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier,PubliclyAccessible]'
+   ```
+
+   This command will return a list of all clusters along with their PubliclyAccessible status.
+
+4. Automate the process: If you have a large number of clusters, you can automate this process by writing a Python script that uses the AWS SDK (Boto3) to list all clusters and check their PubliclyAccessible attribute. The script would essentially do the same thing as the above CLI commands, but in an automated manner.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including Redshift.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a Redshift client using your AWS credentials.
+
+```python
+import boto3
+
+aws_access_key_id = 'YOUR_ACCESS_KEY'
+aws_secret_access_key = 'YOUR_SECRET_KEY'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>